composer/optim/scheduler.py

```diff
@@ -537,30 +537,39 @@ def _raise_if_max_duration_exceeds_t_max(t_max: Union[str, Time], state: State):
         max_dur = Time.from_timestring(max_dur)
 
     max_dur_exceeds_t_max = False
-    if t_max.unit == max_dur.unit and t_max.value < max_dur.value:
-        max_dur_exceeds_t_max = True
-    elif (
-        t_max.unit == TimeUnit.BATCH and max_dur.unit == TimeUnit.EPOCH and state.dataloader_len is not None and
-        t_max.value < max_dur.value * int(state.dataloader_len)
-    ):
-        max_dur_exceeds_t_max = True
-    elif (
-        t_max.unit == TimeUnit.EPOCH and max_dur.unit == TimeUnit.BATCH and state.dataloader_len is not None and
-        t_max.value * int(state.dataloader_len) < max_dur.value
-    ):
-        max_dur_exceeds_t_max = True
-    elif t_max.unit != max_dur.unit:
-        log.info(
-            f'Since max_duration {max_dur} with units {max_dur.unit} and t_max {t_max} with units {t_max.unit} are not '
-            'comparable, make sure that your LR schedule is defined at all points in the training duration.',
-        )
+    if t_max.unit == max_dur.unit:
+        if t_max.value >= max_dur.value:
+            # Time units are comparable, and t_max is valid.
+            return
+        else:
+            max_dur_exceeds_t_max = True
+    elif (t_max.unit == TimeUnit.BATCH and max_dur.unit == TimeUnit.EPOCH and state.dataloader_len is not None):
+        if t_max.value >= max_dur.value * int(state.dataloader_len):
+            # Batches are comparable to epochs through the dataloader length, and t_max is valid.
+            return
+        else:
+            max_dur_exceeds_t_max = True
+    elif (t_max.unit == TimeUnit.EPOCH and max_dur.unit == TimeUnit.BATCH and state.dataloader_len is not None):
+        if t_max.value * int(state.dataloader_len) >= max_dur.value:
+            # Batches are comparable to epochs through the dataloader length, and t_max is valid.
+            return
+        else:
+            max_dur_exceeds_t_max = True
 
     if max_dur_exceeds_t_max:
+        # None of the checks above passed. Time units are comparable, but t_max is invalid since it's less than max_dur.
         raise ValueError(
             f't_max {t_max} must be greater than or equal to max_duration {max_dur}. Otherwise, the LR schedule will '
             'not be defined for the entire training duration.',
         )
 
+    if t_max.unit != max_dur.unit:
+        # Units are not comparable, so we cannot check if t_max is valid. Log this and return.
+        log.debug(
+            f'Since max_duration {max_dur} with units {max_dur.unit} and t_max {t_max} with units {t_max.unit} are not '
+            'comparable, make sure that your LR schedule is defined at all points in the training duration.',
+        )
+
```

Comment:
I agree with Antoine -- the problem described in the issue is less the warning level and more the frequency of logging

Reply:
I think the log was still an error because the durations were equal through the dataloader length even the units were different. The current refactoring solve this with the early returns.

Now that I give it a second though, I'm not sure my proposal of handling it in the `__init__` makes sense because the `max_duration` can change if the `.fit()` if called multiple times.